### PR TITLE
PIME-11: Implement and Test API Call Function

### PR DESCRIPTION
**Pull request by lgaleana-llm**
This PR implements the function to make the API call to the Tequila API from Kiwi and includes unit tests for this function. The function takes in the search parameters as arguments and returns the response from the API. The tests mock the `requests.get` function to return a predefined response, and then check if our function correctly processes this response.

### Test Plan

pytest

### CODE REVIEW

File: api.py [hunk 1]
```diff
@@ -0,0 +1,6 @@
+import requests
```
Comment (lgaleana) [inline api.py:1]: Add a main.py

File: main.py [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import api
```
Comment (lgaleana) [inline main.py:2]: main for FastAPI

File: main.py [hunk 3]
```diff
@@ -0,0 +1,11 @@
+from fastapi import FastAPI
+import api
+
+app = FastAPI()
+
+
+@app.get('/search')
+def search_flights():
+    params = {}
```
Comment (lgaleana) [inline main.py]: How do we get these params?

File: main.py [hunk 4]
```diff
@@ -0,0 +1,10 @@
+from fastapi import FastAPI, Query
+import api
+
+app = FastAPI()
+
+
+@app.get('/search')
+def search_flights(params: dict = Query({})):
```
Comment (lgaleana) [inline main.py]: Be more specific about the params that you need. Something very simple

File: main.py [hunk 5]
```diff
@@ -0,0 +1,11 @@
+from fastapi import FastAPI, Query
+import api
+
+app = FastAPI()
+
+
+@app.get('/search')
+def search_flights(from_city: str = Query(None), to_city: str = Query(None), date: str = Query(None)):
```
Comment (lgaleana) [inline main.py]: Make them mandatory

File: main.py [hunk 6]
```diff
@@ -0,0 +1,11 @@
+from fastapi import FastAPI, Query
+import api
+
+app = FastAPI()
+
+
+@app.get('/search')
+def search_flights(from_city: str = Query(...), to_city: str = Query(...), date: str = Query(...)):
```
Comment (lgaleana) [inline main.py:8]: What does `Query(...)` mean?

Comment (lgaleana-llm) [reply main.py:8]: In FastAPI, `Query(...)` is used to declare additional validation and metadata for HTTP query parameters. The ellipsis `...` is used to make the parameter required. So, `from_city: str = Query(...)` means that the `from_city` query parameter is required and should be of type `str`. If the parameter is not provided in the request, FastAPI will return a 422 Unprocessable Entity response detailing the issue.

**Review by lgaleana — APPROVED** [review]

**Pull request merged**